### PR TITLE
Use 0.4.1 of ThreadingUtilities to stop crashes when !isbits non-array args are used

### LINKED ISSUE #4: # TODO: why does this crash Julia?
Hey, loving the idea of this package and have been playing around with it today. Appreciate it's pretty new and in an early state, but it will be really useful for me.

I've been running into the same error that the commented-out test case has, and it seems to be something to do with the argument type. The only working types I've found are `Number`s and `Array{<:Number}`s. `String`s, `structs and `Array{Array{<:Number}}` cause Julia to crash. 

i.e. this works:
```Julia
function slow_task!((x, digits, n), j, k)
        start = 1 + (n * (j - 1)) ÷ num_threads()
        stop =  (n* k) ÷ num_threads()
        target = 0.0
        for i ∈ start:stop
            target += occursin(string(digits), string(i)) ? 0.0 : 1.0 / i
        end
        x[1,j] = target
    end
    function slow_cheap(n, digits)
        x = zeros(8, num_threads())
        batch(slow_task!, (num_threads(), num_threads()), x, digits, n)
        sum(@view(x[1,1:end]))
    end
    slow_cheap(1000, 9)
```
And this doesn't:
```
function slow_task!((x, digits, n), j, k)
        start = 1 + (n * (j - 1)) ÷ num_threads()
        stop =  (n* k) ÷ num_threads()
        target = 0.0
        for i ∈ start:stop
            target += occursin(digits, string(i)) ? 0.0 : 1.0 / i
        end
        x[1,j] = target
    end
    function slow_cheap(n, digits)
        x = zeros(8, num_threads())
        batch(slow_task!, (num_threads(), num_threads()), x, digits, n)
        sum(@view(x[1,1:end]))
    end
    slow_cheap(1000, "9")
```
Hope this helps debugging this issue, if you weren't already aware of this. Let me know if I can help with this at all!

### PR DESCRIPTION
**Pull request by mmiller-max**
Closes #4

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -25,22 +25,34 @@ using Test
...
+            target += occursin(digits, string(i)) ? 0.0 : 1.0 / i
+        end
+        x[1,j] = target
+    end
+    function slow_cheap(n, digits)
+        x = zeros(8, num_threads())
+        batch(slow_task!, (num_threads(), num_threads()), x, digits, n)
+        sum(@view(x[1,1:end]))
+    end
+    slow_cheap(1000, "9")
```
Comment (chriselrod) [inline test/runtests.jl:42]: Should probably use `@test` (vs a non-threaded version)

**Comment by chriselrod** [comment]
I confirmed that it runs locally. I'll merge it and release once `slow_cheap` is tested to return the correct answer.

File: test/runtests.jl (hunk 1 above)
Comment (mmiller-max) [reply test/runtests.jl:42]: Yep good point!

**Comment by mmiller-max** [comment]
Out of interest, what is causing the failure here? I don't have the permissions to see the details

**Comment by chriselrod** [comment]
> Out of interest, what is causing the failure here? I don't have the permissions to see the details

https://twitter.com/BayesWatch/status/1372674712757747715/photo/1

The only message I have from GitHub so far is support saying they'd ask the actions team about it, so I'm still totally in the dark about what is going on myself. Too much CI use is my best guess, but I thought they were supposed to warn you about it first.

File: test/runtests.jl (hunk 1 above)
Comment (chriselrod) [reply test/runtests.jl]:
```julia
  Expression: slow_cheap(1000, "9") ≈ slow_single_thread(1000, "9")
   Evaluated: 6.589720190283038 ≈ 6.590720190283038
```

Comment (chriselrod) [reply test/runtests.jl]: I wrote the code, so the but is probably my fault.

Comment (mmiller-max) [reply test/runtests.jl]: That's strange, I'm getting:
 ```Julia
Julia> @test slow_cheap(1000, "9") ≈ slow_single_thread(1000,"9")
Test Passed
julia> slow_cheap(1000, "9")
6.5907201902830375

julia> slow_single_thread(1000,"9")
6.590720190283038
```

Comment (mmiller-max) [reply test/runtests.jl]: Oh I've got a change in my local copy. I was gonna do it separately but can include here - the start value is wrong here
https://github.com/JuliaSIMD/CheapThreads.jl/blob/b02d71a5765ff568fe7e56e9ed5e69b638ce2c4c/src/batch.jl#L76
It needs a plus one, as per
https://github.com/JuliaSIMD/CheapThreads.jl/blob/b02d71a5765ff568fe7e56e9ed5e69b638ce2c4c/src/batch.jl#L8

Comment (chriselrod) [reply test/runtests.jl]:
```julia
julia> slow_single_thread(cld(35*1000,36),"9")
6.589720190283038

julia> slow_cheap(1000,"9")
6.589720190283038

julia> Sys.CPU_THREADS
36
```

Comment (chriselrod) [reply test/runtests.jl]: Great!
Just pulled your last commit;
```julia
julia> slow_cheap(1000,"9")
6.590720190283038

julia> slow_single_thread(1000,"9")
6.590720190283038
```

Comment (mmiller-max) [reply test/runtests.jl]: Awesome. I can add an extra test for the values of start and stop if you like?

**Comment by chriselrod** [comment]
Mind bumping the version number?
Then I'll merge and make a release.

(Which I also have to do manually, because no GitHub actions >_<)

File: test/runtests.jl (hunk 1 above)
Comment (chriselrod) [reply test/runtests.jl]: More tests the merrier.

**Comment by chriselrod** [comment]
Great, thanks!

**Comment by mmiller-max** [comment]
No worries!

**Pull request merged**